### PR TITLE
Sp4 forgot password view

### PR DESCRIPTION
**Pull request by Spellbreeze**
Collaborator(s): Xander, Parkaranjot

### CODE REVIEW

File: Civility Optics/Views/Onboarding/ForgotPasswordView.swift [hunk 1]
```diff
@@ -0,0 +1,139 @@
+//
```
Comment (madhura112) [inline Civility Optics/Views/Onboarding/ForgotPasswordView.swift:1]: Everything looks good! Good job!

**Review by madhura112 — APPROVED** [review]
looks good to go!

File: Civility Optics/Views/Onboarding/ForgotPasswordView.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                                .foregroundColor(.stone)
+                                .keyboardType(.alphabet)
+                                .autocapitalization(.none)
+                            Spacer()
+                        }
+                        .padding(.horizontal)
+                    }.frame(width: 300, height: 40)
+                    Button {
+                        model.forgotPassword(email:forgotEmail)
+                        self.sent = true
```
Comment (parkaransingh) [inline Civility Optics/Views/Onboarding/ForgotPasswordView.swift:41]: We may want to return something from the backend to determine if email was actually sent to.

**Review by parkaransingh — APPROVED** [review]
LGTM

File: Civility Optics/Networking/NetworkingService.swift [hunk 3]
```diff
@@ -137,7 +137,70 @@ extension NetworkingService {
         printResponse(data)
       }.resume()
     }
+  
+  static func sendForgotPasswordCode(
```
Comment (menelik-g) [inline Civility Optics/Networking/NetworkingService.swift:141]: Function looks good to me, it also operates correctly

File: Civility Optics/ViewModels/RegisterViewModel.swift [hunk 4]
```diff
@@ -27,6 +27,18 @@ func verifyCode(code: String) {
     }
   }
 
+func forgotPassword(email: String) {
```
Comment (menelik-g) [inline Civility Optics/ViewModels/RegisterViewModel.swift:30]: Function accesses networkingservice, and communicates with database correctly

File: Civility Optics/ViewModels/RegisterViewModel.swift [hunk 5]
```diff
@@ -27,6 +27,18 @@ func verifyCode(code: String) {
...
+    NetworkingService.sendForgotPasswordCode(email: email)
+}
+
+func updateForgottenPassword(email: String, code: String, newPassword: String) {
```
Comment (menelik-g) [inline Civility Optics/ViewModels/RegisterViewModel.swift:34]: function to update password forgotten looks good!

File: Civility Optics/Views/VenueDetails.swift [hunk 6]
```diff
@@ -99,7 +99,6 @@ var email: String
                     // Create Date Formatter
                     Text("Rating: ")
                         .bold()
-//                        .multilineTextAlignment(.leading)
```
Comment (menelik-g) [inline Civility Optics/Views/VenueDetails.swift:102]: This line can be removed if not used

File: Civility Optics/Views/Onboarding/LoginView.swift [hunk 7]
```diff
@@ -13,55 +13,64 @@ struct LoginView: View {
...
-      VStack(alignment: .leading, spacing: 4) {
-        Text("Email")
-          .foregroundColor(.pale)
-          .fontWeight(.semibold)
-        ZStack {
-          RoundedRectangle(cornerRadius: 12)
-            .foregroundColor(.pale)
-          HStack {
-            TextField("Search", text: $email)
-              .foregroundColor(.stone)
```
Comment (menelik-g) [inline Civility Optics/Views/Onboarding/LoginView.swift:33]: two bodies, not sure if both are needed?

**Review by menelik-g — APPROVED** [review]

**Pull request merged**